lesson02/fibonacci/fibonacci.go

```diff
@@ -2,39 +2,33 @@ package fibonacci
 
 import (
 	"fmt"
-	"strings"
+	"log"
 )
 
 func Printer(n int) {
-	one := 0
-	two := 1
-	fmt.Print(one, " ", two, " ")
-	for n > 1 {
-		fib := one + two
-		one, two = two, fib
-		n--
-		fmt.Print(fib, " ")
+	mp := recursive(n)
+	for _, val := range mp {
+		fmt.Printf("%d ", val)
 	}
-	fmt.Println()
 }
 
-func Recursive(n int) {
+func recursive(n int) []int {
 	if n < 0 {
-		fmt.Println("Number must be positive")
-		return
+		log.Fatal("number must be positive")
```

Comment:
I would expect that to be in Printer function, since it's purpose is to output the result of the Fibonacci function. To make it more clean, it would be better to return and error on your `recursive` function along with the slice and then in `Printer` function print out the error if it is not nil.
Also, a minor issue, but still should probably handled. Your wording here is "Number must be positive", but in fact you return this on only non-negative numbers, since 0 is niether positive or negative. Should probably be "Number cannot be negative"